### PR TITLE
chore!: update rust toolchain (proc_macro_is_available feature)

### PR DESCRIPTION
**Pull request by sdbondi**
Checking if CI works

### CODE REVIEW

**Comment by delta1** [comment]
That test failure is very strange

**Comment by sdbondi** [comment]
Yeah very strange, returns an error code for utf8 conversion for the string "msg" - works fine in locally on my mac. Trying on Linux and will see if I can reproduce there.

**Comment by sdbondi** [comment]
Get's even stranger, but on Linux in release the string must be null terminated `"msg\0" - this is only required on release mode on linux.

File: .github/workflows/clippy-check.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           components: clippy, rustfmt
+          toolchain: nightly-2022-01-17
```
Comment (mikethetike) [inline .github/workflows/clippy-check.yml:11]: Don't think this is needed

**Review by mikethetike — APPROVED** [review]

**Pull request merged**